fiaas_deploy_daemon/deployer/kubernetes/deployment/deployer.py

```diff
@@ -94,23 +94,23 @@ def deploy(self, app_spec, selector, labels, besteffort_qos_is_required):
         pod_metadata = ObjectMeta(name=app_spec.name, namespace=app_spec.namespace, labels=pod_labels,
                                   annotations=app_spec.annotations.pod)
         pod_template_spec = PodTemplateSpec(metadata=pod_metadata, spec=pod_spec)
-        replicas = app_spec.replicas
-        # we must avoid that the deployment scales up to app_spec.replicas if autoscaler has set another value
+        replicas = app_spec.autoscaler.min_replicas
+        # we must avoid that the deployment scales up to app_spec.autoscaler.min_replicas if autoscaler has set another value
         if should_have_autoscaler(app_spec):
             try:
                 deployment = Deployment.get(app_spec.name, app_spec.namespace)
                 # the autoscaler won't scale up the deployment if the current number of replicas is 0
                 if deployment.spec.replicas > 0:
                     replicas = deployment.spec.replicas
                     LOG.info("Configured replica size (%d) for deployment is being ignored, as current running replica size"
-                             " is different (%d) for %s", app_spec.replicas, deployment.spec.replicas, app_spec.name)
+                             " is different (%d) for %s", app_spec.autoscaler.min_replicas, deployment.spec.replicas, app_spec.name)
             except NotFound:
                 pass
 
         deployment_strategy = DeploymentStrategy(
             rollingUpdate=RollingUpdateDeployment(maxUnavailable=self._max_unavailable,
                                                   maxSurge=self._max_surge))
-        if app_spec.replicas == 1 and app_spec.singleton:
+        if app_spec.autoscaler.min_replicas == 1 and app_spec.singleton:
```

Comment:
This doesn't look right: before, if 'replicas' was 1, we knew that min_replicas would also have been 1 (max can't be less than min), but here you could have min=1, max=5, and in that case singleton should be ignored, as it doesn't make sense to set it. So I think this should be checking max_replicas. (singleton defaults to true, so this is important - it's not just to deal with the case where a user could set something that doesn't make sense)

Reply:
I agree 👍

Reply:
Fixed in https://github.com/fiaas/fiaas-deploy-daemon/pull/91/commits/2ad9bab78d2e30a411424f1da6f54d37db418449